internal/query/zedtokens.go

```diff
@@ -19,48 +15,49 @@ const (
 	consistencyAtLeastAsFresh  = "at_least_as_fresh"
 )
 
-// initZedTokenCache creates a new LRU cache that watches KV for ZedToken updates.
-func (e *engine) initZedTokenCache() error {
-	status, err := e.kv.Status()
-	if err != nil {
-		return err
+func (e *engine) cacheZedTokens(resp *pb.WatchResponse) {
+	zedToken := resp.ChangesThrough.Token
+
+	for _, update := range resp.Updates {
+		relationship := update.Relationship
+
+		e.zedTokenCache.Add(
+			relationship.Resource.ObjectId,
+			zedToken,
+		)
+
+		e.zedTokenCache.Add(
+			relationship.Subject.Object.ObjectId,
+			zedToken,
+		)
 	}
+}
 
-	ttl := status.TTL()
+// initZedTokenCache creates a new LRU cache that watches SpiceDB for ZedToken updates.
+func (e *engine) initZedTokenCache(ctx context.Context) error {
+	ttl := time.Minute
```

Comment:
Note to self: we should make this configurable based on the revision quantization interval.